src/disco/shred/fd_shred_tile.c

```diff
@@ -0,0 +1,694 @@
+
+    fd_shredder_init_batch( ctx->shredder, entry_batch, entry_batch_sz, entry_batch_meta );
+
+    /* We're depending on the pack tile to produce microblocks that can
+       fit in 1 FEC set. */
+    FD_TEST( fd_shredder_next_fec_set( ctx->shredder, ctx->args->shred_signing_key, out ) );
+    fd_shredder_fini_batch( ctx->shredder );
+
+    d_rcvd_join( d_rcvd_new( d_rcvd_delete( d_rcvd_leave( out->data_shred_rcvd   ) ) ) );
+    p_rcvd_join( p_rcvd_new( p_rcvd_delete( p_rcvd_leave( out->parity_shred_rcvd ) ) ) );
+
+    ctx->send_fec_set_idx = ctx->shredder_fec_set_idx;
+
+    ctx->last_frag_was_shred = 0;
+  } else {
+    /* The FEC resolver API does not present a prepare/commit model. If we
+       get overrun between when the FEC resolver verifies the signature
+       and when it stores the local copy, we could end up storing and
+       retransmitting garbage.  Instead we copy it locally, sadly, and
+       only give it to the FEC resolver when we know it won't be overrun
+       anymore. */
+    uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->net_wksp, chunk );
+    fd_memcpy( ctx->shred_buffer, dcache_entry, sz );
+    ctx->shred_buffer_sz = sz;
+    ctx->last_frag_was_shred = 1;
+  }
+}
+static inline void
+send_shred( shred_tile_ctx_t    * ctx,
+            fd_shred_t const    * shred,
+            fd_shred_dest_idx_t   dest_idx ) {
+  fd_shred_dest_weighted_t * dest = fd_shred_dest_idx_to_dest( ctx->sdest, dest_idx );
+  //FD_LOG_NOTICE(( "Trying to send shred to idx %hx. IP is %x", dest_idx, dest->ip4 ));
+  if( FD_UNLIKELY( !dest->ip4 ) ) return;
+  //FD_LOG_NOTICE(( "Here %i", !dest->ip4 ));
+
+  uchar * packet = fd_chunk_to_laddr( ctx->net_wksp, ctx->net_chunk );
+
+  int is_data = fd_shred_type( shred->variant )==FD_SHRED_TYPE_MERKLE_DATA;
+  eth_ip_udp_t * tmpl = fd_ptr_if( is_data, (eth_ip_udp_t *)ctx->data_shred_net_hdr,
+                                            (eth_ip_udp_t *)ctx->parity_shred_net_hdr );
+  fd_memcpy( packet, tmpl, sizeof(eth_ip_udp_t) );
+
+  eth_ip_udp_t * hdr = (eth_ip_udp_t *)packet;
+
+  memcpy( hdr->eth->dst, dest->mac_addr, 6UL );
+
+  memcpy( hdr->ip4->daddr_c, &dest->ip4, 4UL );
+  hdr->ip4->net_id     = fd_ushort_bswap( ctx->net_id++ );
+  hdr->ip4->check      = 0U;
+  hdr->ip4->check      = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *) FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+
+  hdr->udp->net_dport  = fd_ushort_bswap( dest->port );
+
+  ulong shred_sz = fd_ulong_if( is_data, FD_SHRED_MIN_SZ, FD_SHRED_MAX_SZ );
+  fd_memcpy( packet+sizeof(eth_ip_udp_t), shred, shred_sz );
+
+  ulong pkt_sz = shred_sz + sizeof(eth_ip_udp_t);
+
+  // FD_LOG_HEXDUMP_INFO(( "shred packet", packet, pkt_sz ));
+
+  ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+  ulong   ctl = fd_frag_meta_ctl( ctx->tile_idx, 1, 1, 0 );
+  ulong   sig = fd_disco_netmux_sig( dest->ip4, dest->port, SRC_TILE_SHRED, SRC_TILE_NET );
```

Comment:
As above, the last arg here is not a SRC_TILE_*, it's the idx of the target tile kind that it should go to for round robin selection. The net tile ignores this field and round robins based on seq number for now, so it should be set to 0.

In future we probably want shreds to the same dest to go out on the same net tile, so that's where we would set this and have net tile respect it.